pkg/test/executor/test_runner.go

```diff
@@ -38,69 +38,57 @@ func RunTestCase(
 	require.NoError(t, err)
 	require.True(t, isInstalled)
 
-	prepareStorage := func(getStorage scenario.SetupStorage) []model.StorageSpec {
+	prepareStorage := func(getStorage scenario.SetupStorage) []*models.InputSource {
 		if getStorage == nil {
-			return []model.StorageSpec{}
+			return []*models.InputSource{}
 		}
 
 		storageList, stErr := getStorage(ctx)
 		require.NoError(t, stErr)
 
-		for _, storageSpec := range storageList {
-			inputSource, err := legacy.FromLegacyStorageSpecToInputSource(storageSpec)
+		for _, input := range storageList {
+			strger, err := stack.Nodes[0].ComputeNode.Storages.Get(ctx, input.Source.Type)
 			require.NoError(t, err)
 
-			strger, err := stack.Nodes[0].ComputeNode.Storages.Get(ctx, storageSpec.StorageSource.String())
-			require.NoError(t, err)
-			hasStorage, stErr := strger.HasStorageLocally(ctx, *inputSource)
+			hasStorage, stErr := strger.HasStorageLocally(ctx, *input)
 			require.NoError(t, stErr)
+
 			require.True(t, hasStorage)
 		}
 
 		return storageList
 	}
 
-	spec.Inputs = prepareStorage(testCase.Inputs)
-	spec.Outputs = testCase.Outputs
-	spec.PublisherSpec = model.PublisherSpec{
-		Type: model.PublisherLocal,
-	}
-	spec.Deal = model.Deal{Concurrency: testNodeCount}
-
-	job := model.Job{
-		Metadata: model.Metadata{
-			ID:        "test-job",
-			ClientID:  "test-client",
-			CreatedAt: time.Now(),
-			Requester: model.JobRequester{
-				RequesterNodeID: "test-owner",
-			},
-		},
-		Spec: spec,
-	}
-
-	newJob, err := legacy.FromLegacyJob(&job)
-	require.NoError(t, err)
-	execution := mock.ExecutionForJob(newJob)
-	execution.AllocateResources(newJob.Task().Name, models.Resources{})
-
-	// Since we are submitting jobs directly to the executor, we need to
-	// transform wasm storage specs, which is currently handled by the
-	// requester's endpoint
-	jobTransformers := []jobtransform.PostTransformer{
-		jobtransform.NewWasmStorageSpecConverter(),
-	}
-	for _, transformer := range jobTransformers {
-		_, err = transformer(ctx, newJob)
-		require.NoError(t, err)
-	}
+	job.Task().InputSources = prepareStorage(testCase.Inputs)
+	job.Task().ResultPaths = testCase.Outputs
+	job.Task().Publisher = publisher_local.NewSpecConfig()
+	job.Count = testNodeCount
+
+	// TODO(forrest) [fixme]: we probably need more randomness in the ID to avoid collisions with other tests
+	now := time.Now().UnixNano()
+	job.ID = fmt.Sprintf("test-job-%d", now)
+	job.Meta = make(map[string]string, 2)
+	job.Meta[models.MetaRequesterID] = "test-owner"
+	job.Meta[models.MetaClientID] = "test-client"
+	job.State = models.NewJobState(models.JobStateTypeUndefined)
+	job.Version = 1
+	job.Revision = 1
+	job.CreateTime = now
+	job.ModifyTime = now
+
+	job.Normalize()
+	require.NoError(t, job.Validate())
+
+	execution := mock.ExecutionForJob(job)
+	// TODO(forrest) [fixme]: if we are allocating resources based on name, probably need more randomness
+	execution.AllocateResources(job.Task().Name, models.Resources{})
```

Comment:
task name but within job. if we randomize job ids, then these can be fixed

Reply:
For this test case the jobID is random. But `Task.Name` doesn't take into account the job ID - its a field users provide. I guess if the mapping of execution to task is 1-1 then we need not worry.